mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -340,6 +340,23 @@ def lookup(self, kwargs: TYPE_nkwargs, force_fallback: bool = False) -> Dependen
                         self._subproject_impl(subp_name, varname)
                     break
 
+                # Check if we could have a matching Rust crate imported from
+                # Cargo.lock, with expected api version. We know there is no
+                # wrap-file defined to match this crate.
+                if name.endswith('-rs') and required:
+                    s = name.split('-')
+                    if len(s) < 3:
+                        continue
+                    s.pop() # -rs
+                    s.pop() # version
+                    crate = '-'.join(s)
+                    cargo_wraps = self.wrap_resolver.get_cargo_wraps(crate)
+                    if not cargo_wraps:
+                        continue
+                    raise DependencyException(f'Rust dependency "{name}" not found, ' +
```

Comment:
You don't need the `+` to concatenate string literals.

Reply:
Without `+`, the result is not the expected one (it repeats the first line several times).

Reply:
There are three strings joined with +, only the last one is dynamically built via `.join()`. Concatenating the first two could be done without the +, although the peephole optimizer may also just translate it anyway.

Reply:
Was caused by join who operated on the whole string. Extracted imports in a variable, and use formatted string without `+`.